chalicelib/checks/wfr_checks.py

```diff
@@ -1,1448 +1,1456 @@
+            action.description = "Did not complete action due to time limitations"
             break
         try:
-            run_metawfr.run_metawfr(metawfr_uuid, my_auth, verbose=True, sfn=sfn, env=env)
-            action_logs['runs_checked_or_kicked'].append(metawfr_uuid)
+            run_metawfr.run_metawfr(
+                meta_workflow_run_uuid,
+                connection.ff_keys,
+                sfn=sfn,
+                env=env,
+                valid_status=FINAL_STATUS_TO_RUN,
+            )
+            success.append(meta_workflow_run_uuid)
         except Exception as e:
-            action_logs['error'] = str(e)
-            break
-    action.output = action_logs
-    action.status = 'DONE'
+            error[meta_workflow_run_uuid] = str(e)
+    action.output["success"] = success
+    action.output["error"] = error
     return action
 
 
 @check_function()
 def metawfrs_to_checkstatus(connection, **kwargs):
-    """Find metaworkflowruns that may need status-checking
-    - those with final_status running.
-    running means some workflow runs are actively running.
-    """
-    check = CheckResult(connection, 'metawfrs_to_checkstatus')
-    my_auth = connection.ff_keys
+    """Find MetaWorkflowRuns that may require a status check."""
+    check = initialize_check(connection)
     check.action = "checkstatus_metawfrs"
-    check.description = "Find metaworkflow runs that has workflow runs to be status-checked."
-    check.brief_output = []
-    check.summary = ""
-    check.full_output = {}
-    check.status = 'PASS'
-
-    # check indexing queue
-    env = connection.ff_env
-    indexing_queue = ff_utils.stuff_in_queues(env, check_secondary=True)
-
-    if indexing_queue:
-        check.status = 'PASS'  # maybe use warn?
-        check.brief_output = ['Waiting for indexing queue to clear']
-        check.summary = 'Waiting for indexing queue to clear'
-        check.full_output = {}
-        return check
-
-    query = '/search/?type=MetaWorkflowRun' + \
-            ''.join(['&final_status=' + st for st in ['running']])
-    query += ''.join(['&meta_workflow.title=' + mwf for mwf in default_pipelines_to_run])
-    search_res = ff_utils.search_metadata(query, key=my_auth)
-
-    # nothing to run
-    if not search_res:
-        check.summary = 'All Good!'
-        return check
+    check.description = "Find MetaWorkflowRuns with WorkflowRuns to status check."
 
-    metawfr_uuids = [r['uuid'] for r in search_res]
-    metawfr_titles = [r['title'] for r in search_res]
-
-    check.allow_action = True
-    check.summary = 'Some metawfrs may have wfrs to be status-checked.'
-    check.status = 'WARN'
-    msg = str(len(metawfr_uuids)) + ' metawfrs may have wfrs to be status-checked'
+    meta_workflow_run_uuids = []
+    meta_workflow_run_titles = []
+    query = "/search/?type=MetaWorkflowRun&field=uuid&field=title" + "".join(
+        ["&final_status=" + st for st in FINAL_STATUS_TO_CHECK]
+    )
```

Comment:
this seems like its repeated a few times in different checks, so could turn it into a constant